acstools/acszpt.py

```diff
@@ -0,0 +1,329 @@
+                    'F555W', 'F606W', 'F625W', 'F658N',
+                    'F660N', 'F775W', 'F814W', 'F850LP','F892N'],
+
+            'HRC': ['F220W', 'F250W', 'F330W', 'F344N',
+                    'F435W', 'F475W', 'F502N', 'F550M',
+                    'F555W', 'F606W', 'F625W', 'F658N',
+                    'F660N', 'F775W', 'F814W', 'F850LP','F892N'],
+
+            'SBC': ['F115LP', 'F122M', 'F125LP',
+                   'F140LP', 'F150LP', 'F165LP']
+        }
+        self.zpt_table = None
+
+        # Set the private attributes
+        if filt is None:
+            self._url = 'https://acszeropoints.stsci.edu/results_all/?date={}&' \
+                        'detector={}'.format(self.date, self.detector)
+        else:
+            self.filt = filt.upper()
+
+            self._url = 'https://acszeropoints.stsci.edu/results_single/' \
+                        '?date1={0}&detector={1}&{1}_filter={2}'. \
+                format(self.date, self.detector, self.filt)
+
+        self._zpts = OrderedDict()
+        self._response = None
+        self._failed = False
+
+    def _check_inputs(self):
+        """ Check the input to ensure they are valid
+
+        Returns
+        -------
+
+        """
+        valid_date = True
+        valid_detector = True
+        valid_filter = True
+        # Determine the submitted detector is valid
+        if self.detector not in ['HRC','SBC','WFC']:
+            print('Whoops! {} is not a valid detector option.'.
+                  format(self.detector))
+            print('Please choose one of the following:')
+            print('HRC')
+            print('SBC')
+            print('WFC')
+            print('-'*79)
+            valid_detector = False
+
+        # Determine if the submitted filter is valid
+        if self.filt is None:
+            pass
+        elif self.filt not in self.valid_filters[self.detector]:
+            valid_filter = False
+            print('Whoops! {} is not a valid filter for {}'.
+                  format(self.filt, self.detector))
+            print('Please choose one of the following:')
+            for filt in self.valid_filters[self.detector]:
+                print(filt)
+            print('-' * 79)
+
+        # Determine if the submitted date is valid
+        date_check = self._check_date()
+        if isinstance(date_check, ValueError):
+            valid_date = False
+            print('Whoops! {}'.format(date_check))
+
+        # Only continute if all the inputs are valid options
+        if all([valid_date, valid_filter, valid_detector]):
+            return True
+        else:
+            return False
+
+    def _check_date(self, fmt='%Y-%m-%d'):
+        """Convenience method for determing if the input date is valid
+        """
+        min_date = dt.datetime(2002, 3, 7)
+        try:
+            dt_obj = dt.datetime.strptime(self.date, fmt)
+        except ValueError:
+            return ValueError('Date format does not match YYYY-MM-DD')
```

Comment:
Usually an exception is raised, not returned. If you want to return something, I recommend a status code variable rather than Python exception.